MapboxCoreNavigation/MGLMapboxEvents.swift

```diff
@@ -0,0 +1,165 @@
+        modifiedEventDictionary["originalRequestIdentifier"] = nil
+        modifiedEventDictionary["requestIdentifier"] = nil
+        
+        if let location = routeController.locationManager.location {
+            modifiedEventDictionary["lat"] = location.coordinate.latitude
+            modifiedEventDictionary["lng"] = location.coordinate.longitude
+        }
+        
+        if let geometry = session.originalRoute.coordinates {
+            modifiedEventDictionary["originalGeometry"] = Polyline(coordinates: geometry).encodedPolyline
+            modifiedEventDictionary["originalEstimatedDistance"] = round(session.originalRoute.distance)
+            modifiedEventDictionary["originalEstimatedDuration"] = round(session.originalRoute.expectedTravelTime)
+        }
+        if let geometry = session.currentRoute.coordinates {
+            modifiedEventDictionary["geometry"] = Polyline(coordinates: geometry).encodedPolyline
+            modifiedEventDictionary["estimatedDistance"] = round(session.currentRoute.distance)
+            modifiedEventDictionary["estimatedDuration"] = round(session.currentRoute.expectedTravelTime)
+        }
+        
+        modifiedEventDictionary["distanceCompleted"] = round(session.totalDistanceCompleted + routeProgress.distanceTraveled)
+        modifiedEventDictionary["distanceRemaining"] = round(routeProgress.distanceRemaining)
+        modifiedEventDictionary["durationRemaining"] = round(routeProgress.durationRemaining)
+        
+        modifiedEventDictionary["rerouteCount"] = session.numberOfReroutes
+
+        modifiedEventDictionary["volumeLevel"] = Int(AVAudioSession.sharedInstance().outputVolume * 100)
+        modifiedEventDictionary["screenBrightness"] = Int(UIScreen.main.brightness * 100)
+
+        modifiedEventDictionary["batteryPluggedIn"] = UIDevice.current.batteryState == .charging || UIDevice.current.batteryState == .full
+        modifiedEventDictionary["batteryLevel"] = UIDevice.current.batteryLevel >= 0 ? Int(UIDevice.current.batteryLevel * 100) : -1
+        modifiedEventDictionary["applicationState"] = UIApplication.shared.applicationState.telemString
+        
+        //modifiedEventDictionary["connectivity"] = ??
+        
+        return modifiedEventDictionary
+    }
+}
+
+extension UIApplicationState {
+    var telemString: String {
+        get {
+            switch self {
+            case .active:
+                return "Foreground"
+            case .inactive:
+                return "Inactive"
+            case .background:
+                return "Background"
+            }
+            return "Unknown"
+        }
+    }
+}
+
+extension UIDevice {
+    var machine: String {
+        get {
+            var systemInfo = utsname()
+            uname(&systemInfo)
+            let machineMirror = Mirror(reflecting: systemInfo.machine)
+            let identifier = machineMirror.children.reduce("") { identifier, element in
```

Comment:
It's basically the same. Many of these duplicated class extensions are just stopgaps until we can leverage the full standalone telemetry framework.